src/core/exec-invoke.c

```diff
@@ -959,20 +967,18 @@ static int get_supplementary_groups(const ExecContext *c, const char *user,
          * when we are under root and SupplementaryGroups= is empty.
          */
```

Comment:
Not your fault, but the comment is wrong. `enforce_groups` does setgroups only if `ngids > 0`, i.e. this wouldn't drop any supplementary groups